scalingo/data_scalingo_region.go

```diff
@@ -0,0 +1,86 @@
+package scalingo
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/Scalingo/go-scalingo/v5"
+)
+
+func dataSourceScRegion() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceScRegionsRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"display_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"api": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dashboard": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"database_api": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ssh": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceScRegionsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+
+	regions, err := client.RegionsList(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	name, ok := d.Get("name").(string)
+	if !ok || name == "" {
+		return diag.Errorf("name attribute is mandatory")
+	}
+
+	hasSelected := false
+	var selected scalingo.Region
+	for _, v := range regions {
+		if v.Name == name {
+			selected = v
+			hasSelected = true
+			break
+		}
+	}
+
+	if !hasSelected {
+		return diag.Errorf("notification platform '%v' not found", name)
+	}
```

Comment:
question: do we really need the two variables `hasSelected` and `selected`?
```suggestion
	var selected scalingo.Region
	for _, v := range regions {
		if v.Name == name {
			selected = v
			hasSelected = true
			break
		}
	}

	if selected == scalingo.Region{} {
		return diag.Errorf("region '%v' not found", name)
	}
```